### DISCUSSION TITLE
How to add a border to an image ?

### DISCUSSION THREAD

**Discussion by schlebe** [ImageMagick/ImageMagick#5849] [Help] [answered]
I try to add a border to an image using ImageMagick.

I tried following PowerShell commands

```
$Magick = "D:\Program Files\ImageMagick-7.1.0-Q16-HDRI\magick.exe"
& $Magick i.png -fill transparent -bordercolor "black" -border 2 i.png 

#it is the same thing with 
& $Magick i.png -bordercolor "black" -border 10 i.png 

```

When I look the result, I see that BACKCOLOR that was white is now BLACK !!!

What is happening ? Is that a bug ?

I encountered this problem for the first time with ImageMagick-7.0.10-Q16. To be sure that last ImageMagick doesn't correct this error (or mistake), I have installed ImageMagick-7.1.0-Q16-HDRI (64bits) ... and I have same issue.

What is wrong in my command ?

Image before running ImageMagick

![image](https://user-images.githubusercontent.com/16349543/205496470-e0582879-6ffe-4e90-9d7e-19301429b4ba.png)

Image after having run ImageMagick

![image](https://user-images.githubusercontent.com/16349543/205496592-af0f7fd3-0eb4-4d01-8ee0-568f5ae80a0c.png)

Thanks for any help.

**Comment by snibgo** [comment]
> $Magick i.png -fill transparent -bordercolor "black" -border 2 i.png 

When I run that command, I get an added black border, as expected. It does not change white into black.

You say it changes white into black, but the white pixels within the letters have not changed. I suspect finger-trouble.

"-fill transparent" is redundant. It has no effect here, and can be removed.

**Reply by schlebe** [reply]
I have replaced "black" by "red" and I get following image

![image](https://user-images.githubusercontent.com/16349543/205507022-368bf88c-e789-4ac7-bc79-a1f5636fe1be.png)

Used command now is

    & $Magick i.png -bordercolor "red" -border 2 i.png 

Just for information so that others can reproduce issue using same image, to generate **i.png** file I use following PowerShell commands

```
    $Address = "https://api.voo.be/customer/thumbor/nlXTy1rkdbUxvT5keroc7WTzldc=/fit-in/1024x/--/2022/02/la-3.png"
    wget $Address -OutFile image.webp
    & $Magick image.webp i.png

```

**Comment by snibgo** [comment]
The "wget" fails for me:
```
C:\web\im>wget https://api.voo.be/customer/thumbor/nlXTy1rkdbUxvT5keroc7WTzldc=/fit-in/1024x/--/2022/02/la-3.png

--2022-12-04 18:04:29--  https://api.voo.be/customer/thumbor/nlXTy1rkdbUxvT5keroc7WTzldc=/fit-in/1024x/--/2022/02/la-3.png
Resolving api.voo.be (api.voo.be)... 2606:4700::6812:59d, 2606:4700::6812:49d, 104.18.5.157, ...
Connecting to api.voo.be (api.voo.be)|2606:4700::6812:59d|:443... connected.
GnuTLS: The request is invalid.
Unable to establish SSL connection.
```

**Comment by snibgo** [comment, marked as answer]
I downloaded your webp via Firefox, and converted it to a PNG, name i3.png.

That PNG has a *transparent* white background. As the "-border" documentation https://imagemagick.org/script/command-line-options.php#border says, it works by creating an image of the bordercolor, and compositing the original "over" that. So any transparent pixels will become the bordercolor.

Instead of "over", you can use a compose setting of "Src". Then transparency will not be changed.
```
magick i3.png -compose Src -bordercolor Blue -border 10 x.png
```
![x](https://user-images.githubusercontent.com/33812368/205511626-e68b9ef4-43fe-49b5-8b41-3bcb8ae2b0a1.png)

**Reply by schlebe** [reply]
Thanks, now all is working.
I can generate a TV Channel list using downloaded image from Voo site.

My last result is following png file

![image](https://user-images.githubusercontent.com/16349543/205515745-6432700c-3d1d-4f10-8a5d-4720cd2d01c8.png)
